lib/ansible/modules/cloud/vmware/vmware_drs.py

```diff
@@ -0,0 +1,279 @@
+        self.rule_name = module.params['drs_rule_name']
+        self.enabled = module.params['enabled']
+        self.mandatory = module.params['mandatory']
+        self.affinity_rule = module.params['affinity_rule']
+
+        # Sanity check for cluster
+        self.cluster_obj = find_cluster_by_name(content=self.content,
+                                                cluster_name=self.cluster_name)
+        self.vm_obj_list = []
+        if self.cluster_obj is None:
+            self.module.fail_json(msg="Failed to find the cluster %s" % self.cluster_name)
+        # Sanity check for virtual machines
+        self.get_all_vms_info()
+
+    # Getter
+    def get_all_vms_info(self):
+        """
+        Function to get all VM objects using name from given cluster
+
+        """
+        if not self.vm_list:
+            self.module.fail_json(msg="List of virtual machines can not be empty. "
+                                      "Please specify 'vms' parameter")
+        for vm_name in self.vm_list:
+            vm_obj = find_vm_by_id(content=self.content, vm_id=vm_name,
+                                   vm_id_type='vm_name', cluster=self.cluster_obj)
+            if vm_obj is None:
+                self.module.fail_json(msg="Failed to find the virtual machine %s "
+                                          "in given cluster %s" % (vm_name,
+                                                                   self.cluster_name))
+            self.vm_obj_list.append(vm_obj)
+
+    def get_rules_by_cluster(self, cluster_obj=None):
+        """
+        Function to get DRS rules from given cluster
+        Returns: List of DRS rule
+
+        """
+        if cluster_obj is None:
+            cluster_obj = self.cluster_obj
+        rules_list = []
+        for rule in cluster_obj.configuration.rule:
+            rule_obj = dict(rule_name=rule.name, rule_key=rule.key)
+            rules_list.append(rule_obj)
+        return rules_list
+
+    def get_rule_key_by_name(self, rule_name=None):
+        """
+        Function to get a specific DRS rule key by name
+        Args:
+            rule_name: Name of rule
+
+        Returns: Rule Object if found or None
+
+        """
+        if rule_name:
+            rules_list = self.get_rules_by_cluster()
+            for rule in rules_list:
+                if rule['rule_name'] == rule_name:
+                    return rule['rule_key']
+        # No rule found
+        return None
+
+    # Create
+    def create(self):
+        """
+        Function to check if rule exists or not
+        Returns: Changed status and result
+
+        """
+        rule_key = self.get_rule_key_by_name(rule_name=self.rule_name)
+        if rule_key:
+            self.module.exit_json(changed=False, rule_key=rule_key)
+        self.create_rule_spec()
+
+    def create_rule_spec(self):
+        """
+        Function to create DRS rule
+        Returns: Changed status and result
```

Comment:
create_rule_spec use exit_json, it doesn't return anything.

Reply:
ok.